v2/carbs/carbs.go

```diff
@@ -0,0 +1,300 @@
+package carbs
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	bs "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/multiformats/go-multihash"
+
+	pb "github.com/cheggaaa/pb/v3"
+	car "github.com/ipld/go-car"
+	"github.com/ipld/go-car/util"
+	"golang.org/x/exp/mmap"
+)
+
+var errNotFound = bs.ErrNotFound
+
+// Carbs provides a read-only Car Block Store.
+type Carbs struct {
```

Comment:
nit: I think `carbs.BlockStore` would be more intuitive and less repetitive than `carbs.Carbs`, and also more consistent with `carbs.Index`